### DISCUSSION TITLE
k8s seperation of applications on cluster / namespace / something else?

### DISCUSSION THREAD

**Discussion by Staggerlee011** [BretFisher/ama#199] [Q&A]
Hi, 

When your moving / hosting mulitple apps. When should you seperate them out into there own clusters vs using namespaces and other configuration tools?

**Comment by BretFisher** [comment]
Assuming you had a production cluster, test cluster, etc. I would only create multiple clusters for production apps if there was some requirement for doing so. Some reasons for multiple clusters:

- Different customers. If you're hosting other people's containers, you may not want two customers sharing the same cluster.
- Different datacenter regions.
- Security requirements. You might have strict requirements for only processing certain data on locked-down servers, but everything else can run on a less-hardened cluster.
- Separation of management. You might have two admin groups that want their own clusters, rather than sharing one and using RBAC.
